uefi/src/main.rs

```diff
@@ -366,11 +448,22 @@ fn create_page_tables(
 }
 
 fn init_logger(st: &SystemTable<Boot>, config: BootloaderConfig) -> Option<RawFrameBufferInfo> {
-    let gop = st
+    let gop_handle = st
         .boot_services()
-        .locate_protocol::<GraphicsOutput>()
+        .get_handle_for_protocol::<GraphicsOutput>()
         .ok()?;
-    let gop = unsafe { &mut *gop.get() };
+    let mut gop = unsafe {
+        st.boot_services()
+            .open_protocol::<GraphicsOutput>(
+                OpenProtocolParams {
+                    handle: gop_handle,
+                    agent: st.boot_services().image_handle(),
```

Comment:
Is this the correct handle? Or should we use the `Handle` passed to `main_inner`?

Reply:
I verified with the docs again.

Unfortunately the handle is opaque, and does not implement the method needed to compare them, so I couldn't assert.

Reply:
Document comments:
```
    /// Get the [`Handle`] of the currently-executing image.
    pub fn image_handle(&self) -> Handle {
        // Safety:
        //
        // `IMAGE_HANDLE` is only set by `set_image_handle`, see that
        // documentation for more details.
        //
        // Additionally, `image_handle` takes a `&self` which ensures it
        // can only be called while boot services are active. (After
        // exiting boot services, the image handle should not be
        // considered valid.)
```

And :

```
    /// Update the global image [`Handle`].
    ///
    /// This is called automatically in the `main` entry point as part
    /// of [`uefi_macros::entry`]. It should not be called at any other
    /// point in time, unless the executable does not use
    /// [`uefi_macros::entry`], in which case it should be called once
    /// before calling other `BootServices` functions.
```

Reply:
Thanks for clarifying! I think I'd still prefer if we just passed the handle from main as an addiitional argument. It's more obvious this way imo so there is less chance for confusion when we modify this part of the code again in the future.

Reply:
I think it'd be less confusing to go the other way. It also requires less modifications in the future when UEFI features are added (As we wouldn't be passing the handle through the call stack).

So, IMO, maybe it'd be better to go the other way, and stop using handle directly, instead using it from boot services. But that will require more discussion, and obviously I'll stick with which direction you want to go.

Change has been made, as requested. Regardless of path, it's definitely not great to be inconsistent. :)